### PR TITLE
Split PixelBlendMode into PixelColorBlendingMode and PixelAlphaCompositionMode

### PR DESCRIPTION
**Pull request by vpenades**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

This continues the process started with #535 

As described in the issue, GraphicsOptions now has two separated modes for configuring Color Blending and Alpha Composition.

PixelAlphaCompositionMode has "SrcOver" as first value to ensure that a default GraphicsOptions structure will always be SrcOver-Normal which is the mode everybody expects by default.

Now there's a lot more possible combinations, so creating tests for all of them might take a while. Specially considering the test images are in a separate repository.

### CODE REVIEW

File: src/ImageSharp.Drawing/Processing/Processors/Drawing/FillProcessor.cs [hunk 1]
```diff
@@ -102,14 +102,35 @@ protected override void OnFrameApply(ImageFrame<TPixel> source, Rectangle source
...
-                                                                             && solidBrush.Color.ToVector4().W == 1f)
-                       || (this.options.BlenderMode == PixelBlenderMode.Src));
+            solidBrush = this.brush as SolidBrush<TPixel>;
+
+            if (solidBrush == null)
+            {
+                return false;
+            }
+
+            if (this.options.ColorBlendingMode != PixelColorBlendingMode.Normal)
```
Comment (JimBobSquarePants) [inline src/ImageSharp.Drawing/Processing/Processors/Drawing/FillProcessor.cs]: Much more readable! 👍

File: src/ImageSharp.Drawing/Processing/TextGraphicsOptions.cs [hunk 2]
```diff
@@ -32,7 +32,7 @@ public struct TextGraphicsOptions
 
         private float? dpiY;
 
-        private PixelBlenderMode blenderMode;
+        private PixelColorBlendingMode blenderMode;
```
Comment (tocsoft) [inline src/ImageSharp.Drawing/Processing/TextGraphicsOptions.cs]: we should add the alpha blending option in here too.

Comment (vpenades) [reply src/ImageSharp.Drawing/Processing/TextGraphicsOptions.cs]: @tocsoft I'll take a look

**Comment by vpenades** [comment]
@JimBobSquarePants @tocsoft Some of the coverage tests are failing because the DrawImageProcessor methods I added are not being called from anywere because there were some missing methods in DrawImageExtensions.

And looking at [DrawImageExtensions](https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp.Drawing/Processing/DrawImageExtensions.cs) I've noticed the arguments ordering is a bit of a mess....

Some methods have opacity before blending, and others the opposite way.

Should I rearrange the arguments so opacity is always the last?

- float opacity
- PixelColorBlendingMode color, float opacity
- PixelColorBlendingMode color,PixelAlphaCompositionMode alpha, float opacity

**Comment by JimBobSquarePants** [comment]
Nice coverage!

**Pull request merged**